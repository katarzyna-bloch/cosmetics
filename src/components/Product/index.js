import styled from 'styled-components'
import PropTypes from 'prop-types'

const Name = styled.div`
  text-align: center;
`

const Image = styled.img`
  height: 180px;
  padding: 15px 10px 40px;
`

const Product = (props) => ( 
  <div>
    <Name>{props.name}</Name>
    <Image src={props.img} alt='kosmetyk' />
  </div>
)

Product.propTypes = {
  name: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired,
}

export default Product
