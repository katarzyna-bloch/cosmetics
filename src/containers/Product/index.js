import React, { useEffect } from 'react'
import { compose } from 'redux'
import { connect } from 'react-redux'
import { withRouter } from 'react-router'
import PropTypes from 'prop-types'

import ProductComponent from '../../components/Product'
import { requestProduct } from '../../redux/actions'

export const Product = ({ fetchProducts, product, match: { params: { id } } }) => {
  useEffect(() => {
    fetchProducts(id)
  }, [])
  return (
    <ProductComponent {...product} />
  )
}

const mapStateToProps = ({ product }) => ({
  product: product,
})

const mapDispatchToProps = (dispatch) => ({
  fetchProducts: (id) => dispatch(requestProduct(id)),
})

Product.propTypes = {
  fetchProducts: PropTypes.func.isRequired,
  product: PropTypes.object,
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
}

export default compose(
  connect(mapStateToProps, mapDispatchToProps),
  withRouter,
)(Product)
