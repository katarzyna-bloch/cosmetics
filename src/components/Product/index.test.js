import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Product from '.'

it('should render correctly', () => {
  let wrapped = shallow(<Product />)
  expect(toJson(wrapped)).toMatchSnapshot()
})
