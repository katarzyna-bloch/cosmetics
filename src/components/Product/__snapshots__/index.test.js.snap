// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <styled.div />
  <styled.img
    alt="kosmetyk"
  />
</div>
`;
